gnome-initial-setup/pages/network/gis-network-page.c

```diff
@@ -601,17 +611,62 @@ active_connections_changed (NMClient *client, GParamSpec *pspec, GisNetworkPage
   }
 }
 
+static gboolean wait_connected_timeout_cb (gpointer);
+
 static void
 sync_complete (GisNetworkPage *page)
 {
   GisNetworkPagePrivate *priv = gis_network_page_get_instance_private (page);
   gboolean has_device;
+  NMDeviceState device_state;
+  NMConnectivityState connectivity;
   gboolean activated;
+  gboolean connectivity_check_complete;
   gboolean visible;
 
-  has_device = priv->nm_device != NULL;
-  activated = priv->nm_device != NULL
-    && nm_device_get_state (priv->nm_device) == NM_DEVICE_STATE_ACTIVATED;
+  if (priv->nm_device == NULL) {
+    has_device = FALSE;
+    activated = FALSE;
+    connectivity_check_complete = FALSE;
+  } else {
+    has_device = TRUE;
+    device_state = nm_device_get_state (priv->nm_device);
+    connectivity = nm_client_get_connectivity (priv->nm_client);
+    activated = device_state == NM_DEVICE_STATE_ACTIVATED;
+    connectivity_check_complete = connectivity != NM_CONNECTIVITY_NONE;
```

Comment:
Should it not be checking against `NM_CONNECTIVITY_UNKNOWN`? The NM API documentation says that `UNKNOWN` is used when the connectivity check hasn’t run yet.

Reply:
https://gitlab.gnome.org/GNOME/gnome-initial-setup/merge_requests/74#note_711446 – I'm going to drop this patch downstream for now. In my opinion, it is more important that we notice new devices than to preserve this attempt to make the timezone page show up more rarely, pending more thought on how to make this work well.

Reply:
:+1: